pkg/ghactions/ghactions_helpers.go

```diff
@@ -37,39 +40,53 @@ func parseActionFragments(action string) (owner string, repo string, err error)
 	return frags[0], frags[1], nil
 }
 
-func getCheckSumForTag(ctx context.Context, ghcli *github.Client, owner, repo, tag string) (string, error) {
+func getCheckSumForTag(ctx context.Context, restIf interfaces.REST, owner, repo, tag string) (string, error) {
 	path, err := url.JoinPath("repos", owner, repo, "git", "refs", "tags", tag)
 	if err != nil {
 		return "", fmt.Errorf("failed to join path: %w", err)
 	}
 
-	return doGetReference(ctx, ghcli, path)
+	return doGetReference(ctx, restIf, path)
 }
 
-func getCheckSumForBranch(ctx context.Context, ghcli *github.Client, owner, repo, branch string) (string, error) {
+func getCheckSumForBranch(ctx context.Context, restIf interfaces.REST, owner, repo, branch string) (string, error) {
 	path, err := url.JoinPath("repos", owner, repo, "git", "refs", "heads", branch)
 	if err != nil {
 		return "", fmt.Errorf("failed to join path: %w", err)
 	}
 
-	return doGetReference(ctx, ghcli, path)
+	return doGetReference(ctx, restIf, path)
 }
 
-func doGetReference(ctx context.Context, ghcli *github.Client, path string) (string, error) {
-	req, _ := ghcli.NewRequest(http.MethodGet, path, nil)
+func doGetReference(ctx context.Context, restIf interfaces.REST, path string) (string, error) {
+	req, err := restIf.NewRequest(http.MethodGet, path, nil)
+	if err != nil {
+		return "", fmt.Errorf("cannot create REST request: %w", err)
+	}
+
+	resp, err := restIf.Do(ctx, req)
+
+	if resp != nil {
+		defer func() {
+			_ = resp.Body.Close()
+		}()
+	}
 
-	var t *github.Reference
-	resp, err := ghcli.Do(ctx, req, &t)
 	if err != nil && resp.StatusCode != http.StatusNotFound {
-		if err != nil && strings.Contains(err.Error(), "cannot unmarshal array into Go value of type") {
-			// This is a branch, not a tag
-			return "", nil
-		}
 		return "", fmt.Errorf("failed to do API request: %w", err)
 	} else if resp.StatusCode == http.StatusNotFound {
 		// No error, but no tag found
 		return "", nil
 	}
 
+	var t github.Reference
+	err = json.NewDecoder(resp.Body).Decode(&t)
+	if err != nil && strings.Contains(err.Error(), "cannot unmarshal array into Go value of type") {
```

Comment:
How do I test this branch? The code that resolves a branch uses a different URL (heads/ref) which doesn't seem to return an array..

Reply:
One of the last two test scenarios I added hit this. It was either buf-setup-action or download-syft. I can't recall which of the two.

Reply:
yes, it was buf-setup-action, I stepped through the test with a debugger and hit the code I wanted to!